source/slang/slang-ir-operator-shift-overflow.cpp

```diff
@@ -0,0 +1,66 @@
+// slang-ir-operator-shift-overflow.cpp
+#include "slang-ir-operator-shift-overflow.h"
+
+#include "../../slang.h"
+#include "slang-ir.h"
+#include "slang-ir-insts.h"
+#include "slang-ir-layout.h"
+
+namespace Slang {
+
+    class DiagnosticSink;
+    struct IRModule;
+
+    void checkForOperatorShiftOverflowRecursive(
+        IRInst* inst,
+        CompilerOptionSet& optionSet,
+        DiagnosticSink* sink)
+    {
+        if (auto code = as<IRGlobalValueWithCode>(inst))
+        {
+            for (auto block : code->getBlocks())
+            {
+                for (auto opShiftLeft : block->getChildren())
```

Comment:
In the future we mostly likely are going to extend this pass to check for other operators as well. So might be good to turn this into
```
for (auto inst : block->getChildren())
{
    switch(inst->getOp()) { ... }
}
```

Reply:
`inst` is already in use.
I will go with... `opInst`